ta4j-core/src/main/java/org/ta4j/core/BarSeries.java

```diff
@@ -183,7 +183,14 @@ default String getSeriesPeriodDescription() {
      *
      * If a new bar is added to the series such that the number of bars will exceed
      * the maximum bar count, then the FIRST bar in the series is automatically
-     * removed, ensuring that the maximum bar count is not exceeded.
+     * removed, ensuring that the maximum bar count is not exceeded. The indices of
+     * the bar series do not change.
+     * 
+     * <p>
+     * <b>Example:</b> If you fill a BarSeries with 30 bars and then apply a
+     * {@code maximumBarCount} of 10 bars, the first 20 bars will be removed from
+     * the BarSeries. The indices going further from 0 to 29 remain but return the
+     * same bar from 0 to 20. The remaining 9 bars are returned from index 21.
```

Comment:
would it be better to have this at the javadoc of the `getBar(int i)` and/or `setMaximumBarCount` method? Someone using this method may not check private property descriptions.